presto-main-base/src/main/java/com/facebook/presto/server/QuerySessionSupplier.java

```diff
@@ -141,10 +143,23 @@ else if (context.getTimeZoneId() != null) {
         return sessionBuilder;
     }
 
-    private Identity authenticateIdentity(QueryId queryId, SessionContext context)
+    private Identity authenticateIdentity(QueryId queryId, SessionContext context, String query)
     {
+        try {
+            validateQueryToken(accessControl, queryId, context, query);
+
+            if (isSkipPermissionsCheckEnabled(accessControl)) {
+                return context.getIdentity();
+            }
+        }
+        catch (Exception e) {
+            logUnifiedPermissionsCheckFailure();
```

Comment:
we should catch/log the exception directly inside our implementation of validateQueryToken. A different accessControl might implement validateQueryToken and not want to catch exceptions.

Also, unifiedPermissionCheck is an internal implementation detail. It shouldn't leak out here.